snapcraft/internal/repo/_deb.py

```diff
@@ -289,7 +287,16 @@ def get_package_libraries(cls, package_name: str) -> Set[str]:
 
     @classmethod
     def get_package_for_file(cls, file_path: str) -> str:
-        return _run_dpkg_query_search(file_path)
+        try:
+            absolute_file_path = os.path.join(os.path.sep, file_path)
+            logger.debug(f"searching for {absolute_file_path}")
+            return _run_dpkg_query_search(absolute_file_path)
```

Comment:
the signature for _run_dpkg_query_search would need to change to accept a filepath.Path instead of a string, the inner subprocess call should work with no changes.